scripts/pre-commit.sh

```diff
@@ -82,3 +82,50 @@ if [ $SHOULD_FAIL_PRECOMMIT -ne 0 ]; then
 else
   verify_no_included_apikeys
 fi
+
+# CircleCI doesn't let us split a config file into multiple, so we have a pre-commit hook that does it for us
+# so that the file doesn't become unmanageable. 
+concatenate_circleci_configs() {
```

Comment:
Could we use a CircleCI setup job to generate the final CircleCI config file instead? This would avoid the need for a pre-commit hook, and keep the CircleCI-related logic self contained (and running on CI instead of locally). 

[Here's an example for exactly our use case.](https://circleci.com/docs/using-dynamic-configuration/#a-basic-example)

>In this example, a script is used to generate a YAML configuration file. The continuation orb is used to continue the pipeline using the generated configuration.